pyccl/tracers.py

```diff
@@ -0,0 +1,405 @@
+                 and isinstance(dndz[1], collections.Iterable)))):
+        raise ValueError("dndz needs to be a tuple of two arrays.")
+    z_n, n = _check_array_params(dndz)
+    has_magbias = mag_bias is not None
+    z_s, s = _check_array_params(mag_bias)
+
+    # Calculate number of samples in chi
+    nchi = lib.get_nchi_lensing_kernel_wrapper(z_n)
+    # Compute array of chis
+    status = 0
+    chi, status = lib.get_chis_lensing_kernel_wrapper(cosmo.cosmo, z_n[-1],
+                                                      nchi, status)
+    # Compute kernel
+    wchi, status = lib.get_lensing_kernel_wrapper(cosmo.cosmo,
+                                                  z_n, n, z_n[-1],
+                                                  int(has_magbias), z_s, s,
+                                                  chi, nchi, status)
+    check(status)
+    return chi, wchi
+
+
+def get_kappa_kernel(cosmo, z_source, nsamples):
+    """This convenience function returns the radial kernel for
+    CMB-lensing-like tracers.
+
+    Args:
+        cosmo (:obj:`Cosmology`): Cosmology object.
+        z_source (float): Redshift of source plane for CMB lensing.
+        nsamples (int): number of samples over which the kernel
+            is desired. These will be equi-spaced in radial distance.
+            The kernel is quite smooth, so usually O(100) samples
+            is enough.
+    """
+    chi_source = comoving_radial_distance(cosmo, 1./(1.+z_source))
+    chi = np.linspace(0, chi_source, nsamples)
+
+    status = 0
+    wchi, status = lib.get_kappa_kernel_wrapper(cosmo.cosmo, chi_source,
+                                                chi, nsamples, status)
+    check(status)
+    return chi, wchi
+
+
+class Tracer(object):
+    """Tracers contain the information necessary to describe the
+    contribution of a given sky observable to its cross-power spectrum
+    with any other tracer. Tracers are composed of 4 main ingredients:
+
+    * A radial kernel: this expresses the support in redshift/distance
+      over which this tracer extends.
+
+    * A transfer function: this is a function of wavenumber and
+      scale factor that describes the connection between the tracer
+      and the power spectrum on different scales and at different
+      cosmic times.
+
+    * An ell-dependent prefactor: normally associated with angular
+      derivatives of a given fundamental quantity.
+
+    * The order of the derivative of the Bessel functions with which
+      they enter the computation of the angular power spectrum.
+
+    A `Tracer` object will in reality be a list of different such
+    tracers that get combined linearly when computing power spectra.
+    Further details can be found in Section 4.9 of the CCL note.
+    """
+    def __init__(self):
+        """By default this `Tracer` object will contain no actual
+        tracers
+        """
+        # Do nothing, just initialize list of tracers
+        self.trc = []
```

Comment:
We should either make this "private" via prefixing with `_` or make it a public attribute that is documented. I think private is better so it would `self._trc`.

Reply:
Done